tests/integration/test_rest_catalog.py

```diff
@@ -16,34 +16,788 @@
+            ],
+        }
+    ],
+    "properties": {
+        "owner": "fokko",
+        "write.parquet.compression-codec": "zstd",
+    },
+    "refs": {},
+    "snapshots": [],
+    "snapshot-log": [],
+    "metadata-log": [],
+}
+
+
+@pytest.fixture
+def table_metadata_no_snapshot_v2() -> Dict[str, Any]:
+    return EXAMPLE_table_metadata_no_snapshot_v2
+
 
-TEST_NAMESPACE_IDENTIFIER = "TEST NS"
+@pytest.fixture
+def table_metadata_no_snapshot_partitioned_v2() -> Dict[str, Any]:
+    return EXAMPLE_table_metadata_no_snapshot_partitioned_v2
+
+
+@pytest.fixture
+def rest_integration_example_metadata_partitioned_v2(table_metadata_no_snapshot_partitioned_v2: Dict[str, Any]) -> Dict[str, Any]:
+    return {
+        "metadata-location": f"s3://warehouse/{TEST_TABLE_IDENTIFIER[0]}/{TEST_TABLE_IDENTIFIER[1]}",
+        "metadata": table_metadata_no_snapshot_partitioned_v2,
+        "config": {
+            "client.factory": "io.tabular.iceberg.catalog.TabularAwsClientFactory",
+            "region": "us-west-2",
+        },
+    }
+
+
+@pytest.fixture
+def example_table_metadata_with_no_location(table_metadata_no_snapshot_v2: Dict[str, Any]) -> Dict[str, Any]:
+    return {
+        "metadata": table_metadata_no_snapshot_v2,
+        "config": {
+            "client.factory": "io.tabular.iceberg.catalog.TabularAwsClientFactory",
+            "region": "us-west-2",
+        },
+    }
+
+
+@pytest.fixture
+def table_metadata_no_snapshot_rest(table_metadata_no_snapshot_v2: Dict[str, Any]) -> Dict[str, Any]:
+    return {
+        "metadata-location": f"s3://warehouse/{TEST_TABLE_IDENTIFIER[0]}/{TEST_TABLE_IDENTIFIER[1]}",
+        "metadata": table_metadata_no_snapshot_v2,
+        "config": {
+            "client.factory": "io.tabular.iceberg.catalog.TabularAwsClientFactory",
+            "region": "us-west-2",
+        },
+    }
 
 
 @pytest.mark.integration
-@pytest.mark.parametrize("catalog", [pytest.lazy_fixture("session_catalog")])
-def test_namespace_exists(catalog: RestCatalog) -> None:
-    if not catalog.namespace_exists(TEST_NAMESPACE_IDENTIFIER):
+@pytest.fixture(scope="function")
+@pytest.mark.parametrize("catalog", [pytest.lazy_fixture("session_catalog"), pytest.lazy_fixture("test_clean_up")])
+def test_clean_up(catalog: RestCatalog) -> None:
+    for namespaces_tuple in catalog.list_namespaces():
+        namespace_name = namespaces_tuple[0]
+        if TEST_NAMESPACE_IDENTIFIER[0] in namespace_name:
+            for identifier in catalog.list_tables(namespace_name):
+                catalog.purge_table(identifier)
+            if catalog.namespace_exists(TEST_NAMESPACE_IDENTIFIER):
+                catalog.drop_namespace(TEST_NAMESPACE_IDENTIFIER)
+
+
+@pytest.mark.integration
+@pytest.mark.parametrize("catalog,clean_up", [(pytest.lazy_fixture("session_catalog"), pytest.lazy_fixture("test_clean_up"))])
+def test_create_namespace_200(catalog: RestCatalog, clean_up: Any) -> None:
```

Comment:
nit: the naming is confusing since we dont assert the 200 status code. 

how about just `test_create_namespace`